Upgrade to new VMC SDK versions

dimitarproynov opened the pull request:
VMC Backend service's swagger.json has diverged from previous version, from which the VMC SDK was emitted leading to significant differences in the binary in the nsxt-vmc-aws-integration package.

The package was used for the public Ip resource.

Testing done:
make build
make test
golangci-lint run

make testacc TESTARGS="-run='TestAccResourceVmcSddcZerocloud|TestAccResourceVmcClusterZerocloud|TestAccResourceVmcSiteRecoveryZerocloud|TestAccResourceVmcSrmNodeZerocloud|TestAccDataSourceVmcCustomerSubnetsBasic|TestAccDataSourceVmcConnectedAccountsBasic|TestAccDataSourceVmcOrgBasic|TestAccDataSourceVmcSddcBasic|TestAccResourceSddcGroupZerocloud' -parallel 4" Passed

Public IP resource Acceptance test passed against real production env: make testacc TESTARGS="-run='TestAccResourceVmcPublicIp_basic'" TF_ACC=1 go test $(go list ./... |grep -v 'vendor') -v -run='TestAccResourceVmcPublicIp_basic' -timeout 240m
?   	github.com/vmware/terraform-provider-vmc	[no test files]
=== RUN   TestAccResourceVmcPublicIp_basic
Reverse proxy : https://nsx-xx-xxx-xxx-xx.rp.vmwarevmc.com/vmc/reverse-proxy/api/orgs/xxx/sddcs/xxx/sks-nsxt-manager
UUID : 186626c2-xxxx-xxxx-8c9b-d827d0cd5bb7 publicIP : <nil>--- PASS: TestAccResourceVmcPublicIp_basic (12.60s)
PASS
ok  	github.com/vmware/terraform-provider-vmc/vmc	13.665s
?   	github.com/vmware/terraform-provider-vmc/vmc/connector	[no test files]
?   	github.com/vmware/terraform-provider-vmc/vmc/constants	[no test files]
testing: warning: no tests to run
PASS
ok  	github.com/vmware/terraform-provider-vmc/vmc/sddcgroup	(cached) [no tests to run]
testing: warning: no tests to run
PASS
ok  	github.com/vmware/terraform-provider-vmc/vmc/task	(cached) [no tests to run]

Commit:
Upgrade to new VMC SDK versions

VMC Backend service's swagger.json has diverged from previous version, from which the VMC SDK was emitted leading to significant differences in the binary in the nsxt-vmc-aws-integration package.

The package was used for the public Ip resource.

Testing done:
make build
make test
golangci-lint run

make testacc TESTARGS="-run='TestAccResourceVmcSddcZerocloud|TestAccResourceVmcClusterZerocloud|TestAccResourceVmcSiteRecoveryZerocloud|TestAccResourceVmcSrmNodeZerocloud|TestAccDataSourceVmcCustomerSubnetsBasic|TestAccDataSourceVmcConnectedAccountsBasic|TestAccDataSourceVmcOrgBasic|TestAccDataSourceVmcSddcBasic|TestAccResourceSddcGroupZerocloud' -parallel 4" Passed

Public IP resource Acceptance test passed against real production env:
make testacc TESTARGS="-run='TestAccResourceVmcPublicIp_basic'"
TF_ACC=1 go test $(go list ./... |grep -v 'vendor') -v -run='TestAccResourceVmcPublicIp_basic' -timeout 240m
?   	github.com/vmware/terraform-provider-vmc	[no test files]
=== RUN   TestAccResourceVmcPublicIp_basic
Reverse proxy : https://nsx-xx-xxx-xxx-xx.rp.vmwarevmc.com/vmc/reverse-proxy/api/orgs/xxx/sddcs/xxx/sks-nsxt-manager
UUID : 186626c2-xxxx-xxxx-8c9b-d827d0cd5bb7 publicIP : <nil>--- PASS: TestAccResourceVmcPublicIp_basic (12.60s)
PASS
ok  	github.com/vmware/terraform-provider-vmc/vmc	13.665s
?   	github.com/vmware/terraform-provider-vmc/vmc/connector	[no test files]
?   	github.com/vmware/terraform-provider-vmc/vmc/constants	[no test files]
testing: warning: no tests to run
PASS
ok  	github.com/vmware/terraform-provider-vmc/vmc/sddcgroup	(cached) [no tests to run]
testing: warning: no tests to run
PASS
ok  	github.com/vmware/terraform-provider-vmc/vmc/task	(cached) [no tests to run]

Signed-off-by: Dimitar Proynov <proynovd@vmware.com>

dekp commented:
Should also replace the use of `Connector.SetSecurityContext()` in `vmc/connector/clientconnector.go`, similar to what is done for `vmc/sddcgroup/sddc_group_client.go` and `vmc/task/task_v2_client.go`.

Commit:
Fix SDDC Provisioning in SDDC Group Acceptance tests

Instead of provisioning new sddcs for the SDDC Group Acceptance, import them via da sddc data source

Address Deyan's comment about removing deprecated "Connector.SetSecurityContext()" from clientconnector.go.

Testing done:
make build
make test
golangci-lint run

make testacc TESTARGS="-run='TestAccResourceSddcGroupZerocloud'"
TF_ACC=1 go test $(go list ./... |grep -v 'vendor') -v -run='TestAccResourceSddcGroupZerocloud' -timeout 240m
?   	github.com/vmware/terraform-provider-vmc	[no test files]
=== RUN   TestAccResourceSddcGroupZerocloud
=== PAUSE TestAccResourceSddcGroupZerocloud
=== CONT  TestAccResourceSddcGroupZerocloud
--- PASS: TestAccResourceSddcGroupZerocloud (179.81s)
PASS

Signed-off-by: Dimitar Proynov <proynovd@vmware.com>

Commit:
Fix SDDC Provisioning in SDDC Group Acceptance tests

Add the forgoten SDDC_GROUP_TEST_SDDC_1_ID and secrets.SDDC_GROUP_TEST_SDDC_2_ID to the environment setup for acceptance tests.

Testing done:
make build
make test
golangci-lint run

make testacc TESTARGS="-run='TestAccResourceSddcGroupZerocloud'"
TF_ACC=1 go test $(go list ./... |grep -v 'vendor') -v -run='TestAccResourceSddcGroupZerocloud' -timeout 240m
?   	github.com/vmware/terraform-provider-vmc	[no test files]
=== RUN   TestAccResourceSddcGroupZerocloud
=== PAUSE TestAccResourceSddcGroupZerocloud
=== CONT  TestAccResourceSddcGroupZerocloud
--- PASS: TestAccResourceSddcGroupZerocloud (179.81s)
PASS

Signed-off-by: Dimitar Proynov <proynovd@vmware.com>

Commit:
Upgrade golangci to v1.52.2

The golangci-lint runs on GitHub started failing.

Testing done:
make build
make test
golangci-lint run

Signed-off-by: Dimitar Proynov <proynovd@vmware.com>

Commit:
Fix the clientconnector.go to use the Rest protocol, instead of the JSON RPC (vapi)

Testing done:
make build
make test
golangci-lint run

make testacc TESTARGS="-run='TestAccResourceVmcSddcZerocloud|TestAccResourceVmcClusterZerocloud|TestAccResourceVmcSiteRecoveryZerocloud|TestAccResourceVmcSrmNodeZerocloud|TestAccDataSourceVmcCustomerSubnetsBasic|TestAccDataSourceVmcConnectedAccountsBasic|TestAccDataSourceVmcOrgBasic|TestAccDataSourceVmcSddcBasic|TestAccResourceSddcGroupZerocloud' -parallel 4"

All E2E tests pass.

Signed-off-by: Dimitar Proynov <proynovd@vmware.com>

karaatanassov reviewed (commented):
Looks good thank you!
